inlong-sort/sort-connectors/iceberg/src/main/java/org/apache/inlong/sort/iceberg/sink/multiple/IcebergMultipleStreamWriter.java

```diff
@@ -0,0 +1,227 @@
+
+            TaskWriterFactory<RowData> taskWriterFactory = new RowDataTaskWriterFactory(
+                    table,
+                    recordWithSchema.getSchema(),
+                    FlinkSchemaUtil.convert(recordWithSchema.getSchema()),
+                    targetFileSizeBytes,
+                    fileFormat,
+                    equalityFieldIds,
+                    upsertMode,
+                    appendMode);
+
+            if (multipleWriters.get(tableId) == null) {
+                IcebergSingleStreamWriter<RowData> writer = new IcebergSingleStreamWriter<>(
+                        tableId.toString(), taskWriterFactory, null, null);  // todo：后面再考虑metric的兼容
+                writer.setup(getRuntimeContext(),
+                        new CallbackCollector<>(writeResult ->
+                                collector.collect(new MultipleWriteResult(tableId, writeResult))),
+                        context);
+                writer.initializeState(functionInitializationContext);
+                writer.open(new Configuration());
+                multipleWriters.put(tableId, writer);
+            } else {  // only if second times schema will evolute
+                multipleWriters.get(tableId).schemaEvolution(taskWriterFactory);
+            }
+
+        }
+
+        if (multipleWriters.get(tableId) != null) {
+            for (RowData data : recordWithSchema.getData()) {
+                multipleWriters.get(tableId).processElement(data);
+            }
+        } else {
+            LOG.error("Unregistered table schema for {}.", recordWithSchema.getTableId());
+        }
+    }
+
+    @Override
+    public void prepareSnapshotPreBarrier(long checkpointId) throws Exception {
+        for (Entry<TableIdentifier, IcebergSingleStreamWriter<RowData>> entry: multipleWriters.entrySet()) {
+            entry.getValue().prepareSnapshotPreBarrier(checkpointId);
+        }
+    }
+
+    @Override
+    public void snapshotState(FunctionSnapshotContext context) throws Exception {
+        for (Entry<TableIdentifier, IcebergSingleStreamWriter<RowData>> entry: multipleWriters.entrySet()) {
+            entry.getValue().snapshotState(context);
+        }
+    }
+
+    @Override
+    public void initializeState(FunctionInitializationContext context) throws Exception {
+        this.functionInitializationContext = context;
+    }
+
+    private boolean isSchemaUpdate(RecordWithSchema recordWithSchema) {
+        TableIdentifier tableId = recordWithSchema.getTableId();
+        recordWithSchema.replaceSchema();
+        if (multipleSchemas.get(tableId) != null
+                && multipleSchemas.get(tableId).sameSchema(recordWithSchema.getSchema())) {
+            return false;
+        }
+        LOG.info("Schema evolution with table {}, old schema: {}, new Schema: {}",
+                tableId, multipleSchemas.get(tableId), recordWithSchema.getSchema());
+        multipleSchemas.put(recordWithSchema.getTableId(), recordWithSchema.getSchema());

```

Comment:
Why put it in multipleSchemas here?

Reply:
put schema in cache because every time data come into, it should know scheam update or not. If updated ,the writer will be new, and old wrtier shoule close , so it shoule cache schema to compare data schema and previous schema.